dbms/src/Storages/KVStore/MultiRaft/RegionSerde.cpp

```diff
@@ -121,112 +86,82 @@ inline size_t mockInjectExtension(std::optional<std::any> v, UInt32 & actual_ext
             s.data(),
             s.size());
     }
-    if (value & 2)
-    {
-        std::string s = "kkk";
-        total_size
-            += writePersistExtension(actual_extension_count, buf, UNUSED_EXTENSION_NUMBER_FOR_TEST, s.data(), s.size());
-    }
+#endif
     return total_size;
 }
 
-std::tuple<size_t, UInt64> Region::serialize(WriteBuffer & buf) const
+template <UInt32 num_ext>
```

Comment:
This is a very good idea. However may be the template parameter is not necessary. I think we can rely on some compiler opt